cmd/olm/main.go

```diff
@@ -106,6 +123,27 @@ func main() {
 	http.Handle("/metrics", prometheus.Handler())
 	go http.ListenAndServe(":8080", nil)
 
-	_, done := operator.Run(stopCh)
+	ready, done := operator.Run(stopCh)
+	<-ready
+
+	if *writeStatus == true {
+		existing, err := configClient.ClusterOperators().Get("operator-lifecycle-manager", metav1.GetOptions{})
```

Comment:
I'm assuming I don't need to worry about creation, since that should already be done.